### PR TITLE
ARROW-9737: [C++][Gandiva] Add bitwise_xor() for integers

### PR DESCRIPTION
**Pull request by zhztheplayer**
https://issues.apache.org/jira/browse/ARROW-9737

### CODE REVIEW

**Comment by kiszk** [comment]
Looks good to me

File: cpp/src/gandiva/precompiled/arithmetic_ops_test.cc [hunk 1]
```diff
@@ -120,6 +120,14 @@ TEST(TestArithmeticOps, TestBitwiseOps) {
...
   EXPECT_EQ(bitwise_or_int64_int64(0x6A5B1, 0x00000), 0x6A5B1);
 
+  // bitwise XOR
+  EXPECT_EQ(bitwise_xor_int32_int32(0x0147D, 0x17159), 0x16524);
+  EXPECT_EQ(bitwise_xor_int32_int32(0xFFFFFFCC, 0x00000297), 0XFFFFFD5B);
+  EXPECT_EQ(bitwise_xor_int32_int32(0x000, 0x285), 0x285);
+  EXPECT_EQ(bitwise_xor_int64_int64(0x563672F83, 0x0D9FCF85B), 0x5BA9BD7D8);
+  EXPECT_EQ(bitwise_xor_int64_int64(0xFFFFFFFFFFDA8F6A, 0xFFFFFFFFFFFF791C), 0X25F676);
+  EXPECT_EQ(bitwise_xor_int64_int64(0x6A5B1, 0x00000), 0x6A5B1);
+
```
Comment (emkornfield) [inline cpp/src/gandiva/precompiled/arithmetic_ops_test.cc:130]: could you add a test to show xor a value with itself is zero?

Comment (zhztheplayer) [reply cpp/src/gandiva/precompiled/arithmetic_ops_test.cc:131]: done

**Comment by emkornfield** [comment]
+1 thank you.

**Pull request closed without merging**